metric_learn/sdml.py

```diff
@@ -131,5 +132,8 @@ def fit(self, X, y, random_state=np.random):
 
     c = Constraints.random_subset(y, self.num_labeled,
                                   random_state=random_state)
-    adj = c.adjacency_matrix(num_constraints, random_state=random_state)
-    return SDML.fit(self, X, adj)
+    pos_neg = c.positive_negative_pairs(num_constraints,
+                                              random_state=random_state)
+    pairs, y = wrap_pairs(X, pos_neg)
+    y = 2 * y - 1
```

Comment:
shouldn't the pair labels be 0/1 and not -1/1?

that said, this raises the question of whether pair labels should be 0/1 or -1/1. If -1/1 is always more convenient for the algorithm implementation, we could switch to that. potentially allow both and convert to -1/1 in the process pairs helper function

Reply:
Yes I agree
Done (see commits https://github.com/metric-learn/metric-learn/pull/92/commits/374a8512ff13b8c427c03ed03a36a782fa8ef55c and https://github.com/metric-learn/metric-learn/pull/92/commits/b4bdec4bf30875253e6a60e8f6c22863daa2bf7d). (Note that for now I did not try to simplify the code of the algorithms using these labels but I just added it in the new API issue main message)